runtime/src/accounts_index.rs

```diff
@@ -1150,32 +1094,38 @@ impl<T: IndexValue> AccountsIndex<T> {
         // instead of scanning the entire range
         let mut total_elapsed_timer = Measure::start("total");
         let mut num_keys_iterated = 0;
-        let mut latest_slot_elapsed = 0;
         let mut load_account_elapsed = 0;
         let mut read_lock_elapsed = 0;
         let mut iterator_elapsed = 0;
         let mut iterator_timer = Measure::start("iterator_elapsed");
         for pubkey_list in self.iter(range.as_ref(), config.collect_all_unsorted) {
             iterator_timer.stop();
             iterator_elapsed += iterator_timer.as_us();
-            for (pubkey, list) in pubkey_list {
+            for pubkey in pubkey_list {
                 num_keys_iterated += 1;
                 let mut read_lock_timer = Measure::start("read_lock");
-                let list_r = &list.slot_list.read().unwrap();
+                let result = self.get2(&pubkey, Some(ancestors), max_root, |entry| {
+                    entry.map(|(locked_entry, index)| {
+                        let slot_list = locked_entry.slot_list();
+                        slot_list[index]
+                    })
+                });
```

Comment:
Maybe iterator could return this slot list information while it was parsing out the pubkeys, if the iterator took a closure